src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -42,10 +43,13 @@ object RuleUtils {
       val sourcePlanSignature = sourcePlanSignatures.head
 
       if (!signatureMap.contains(sourcePlanSignature.provider)) {
-        val signature = LogicalPlanSignatureProvider
+        LogicalPlanSignatureProvider
           .create(sourcePlanSignature.provider)
-          .signature(plan)
-        signatureMap.put(sourcePlanSignature.provider, signature)
+          .signature(plan) match {
+          case Some(s) =>
+            signatureMap.put(sourcePlanSignature.provider, s)
+          case None => return false
```

Comment:
I think we should still put the value if the signature generation is already tried. You probably need to change the map to `val signatureMap = mutable.Map[String, Option[String]]()`.

Reply:
Makes sense, changed it. Thnx